diesel/src/deserialize.rs

```diff
@@ -55,8 +63,59 @@ pub type Result<T> = result::Result<T, Box<Error + Send + Sync>>;
 /// # }
 /// ```
 ///
-/// If we want to do additional work during deserialization, we can implement
-/// the trait ourselves.
+/// If we want to do additional work during deserialization, we can use
+/// `deserialize_as` to use a different implementation.
+///
+/// ```rust
+/// # #[macro_use] extern crate diesel;
+/// # include!("doctest_setup.rs");
+/// #
+/// # use schema::users;
+/// # use diesel::backend::Backend;
+/// # use diesel::deserialize::Queryable;
+/// #
+/// struct LowercaseString(String);
+///
+/// impl Into<String> for LowercaseString {
+///     fn into(self) -> String {
+///         self.0
+///     }
+/// }
+///
+/// impl<DB, ST> Queryable<ST, DB> for LowercaseString
+/// where
+///     DB: Backend,
+///     String: Queryable<ST, DB>,
+/// {
+///     type Row = <String as Queryable<ST, DB>>::Row;
+///
+///     fn build(row: Self::Row) -> Self {
+///         LowercaseString(String::build(row).to_lowercase())
+///     }
+/// }
+///
+/// #[derive(Queryable, PartialEq, Debug)]
+/// struct User {
+///     id: i32,
+///     #[diesel(deserialize_as = "LowercaseString")]
+///     name: String,
```

Comment:
Huh, from the description I somehow expected this to be the other way around: Deserialize it "like a string", and then call `LowercaseString::from(string)` so you end up with a `LowercaseString` here as type of `name`. I can see now that this would move the whole logic into the `Into` impl and you can't work with the raw row data, though, so you might be converting the data twice (which in this example isn't really necessary).

Reply:
I'm not sure I understand what you're saying. Do you mean that you expected this to be helpful for when you *want* to use the newtype on your struct, requiring only a `From` impl instead of the relevant trait?

The goal of this feature is to let you write impls on newtypes for coherence purposes, but not have to use the newtype on your struct (since they tend to be painful to use). Keep in mind, a lot of this cases this is meant to enable are cases where the inner type isn't supported by Diesel (e.g. all the folks who want `impl FromSql<Binary, Sqlite> for Uuid`)